controllers/progressiverollout_controller_test.go

```diff
@@ -22,6 +25,26 @@ const (
 	interval = time.Millisecond * 10
 )
 
+type mockArgoCDAppClientCounter struct {
+	appsSynced []string
+	m          sync.Mutex
+}
+
+func (c *mockArgoCDAppClientCounter) Sync(ctx context.Context, in *applicationpkg.ApplicationSyncRequest, opts ...grpc.CallOption) (*argov1alpha1.Application, error) {
+	c.m.Lock()
+	c.appsSynced = append(c.appsSynced, *in.Name)
+	defer c.m.Unlock()
+
+	return nil, nil
+}
+
+func (c *mockArgoCDAppClientCounter) GetSyncedApps() []string {
+	c.m.Lock()
+	defer c.m.Unlock()
```

Comment:
What is the lock about here?

Reply:
There's still a read in this method (in the return statement), so we need a read lock too.